cmd/rpc.go

```diff
@@ -19,25 +28,52 @@ const (
 var (
 	requestURL    string
 	authTokenFlag string
+	storePath     string
 )
 
-func InitURLFlag() (*string, string, string, string) {
-	return &requestURL, "url", defaultRPCAddress, "Request URL"
-}
+func RPCFlags() *flag.FlagSet {
+	fset := &flag.FlagSet{}
+
+	fset.StringVar(
+		&requestURL,
+		"url",
+		defaultRPCAddress,
+		"Request URL",
+	)
 
-func InitAuthTokenFlag() (*string, string, string, string) {
-	return &authTokenFlag,
+	fset.StringVar(
+		&authTokenFlag,
 		"token",
 		"",
-		"Authorization token (if not provided, the " + authEnvKey + " environment variable will be used)"
+		"Authorization token (if not provided, the "+authEnvKey+" environment variable will be used)",
+	)
+
+	storeFlag := NodeFlags().Lookup(nodeStoreFlag)
+
+	fset.StringVar(
+		&storePath,
+		nodeStoreFlag,
+		"",
+		storeFlag.Usage,
+	)
+	return fset
 }
 
 func InitClient(cmd *cobra.Command, _ []string) error {
+	var (
+		err   error
+		token string
+	)
+
 	if authTokenFlag == "" {
-		authTokenFlag = os.Getenv(authEnvKey)
+		token, err = getToken(storePath)
+	}
+
+	if err != nil {
+		token = os.Getenv(authEnvKey)
 	}
```

Comment:
IMO env check should be first as that's what's expected / documented, and THEN the token can be grabbed from the store path in case no auth token is set in env var

Reply:
or we could just do away with env var entirely and have it work by grabbing token from store by default and if user needs to specify token, they can do with flag. cc @distractedm1nd

Reply:
I like the second approach but in this case user has to specify either token or the path to the root each time, that could be annoying. I would move the reading from env upper - before getting the token from the config

Reply:
This is moving into devex territory and so would love input from @jcstein 
To sum it up we have two approaches we are considering:
1. We use the token in the CELESTIA_NODE_AUTH_TOKEN, but if it doesnt exist it grabs it from the node store
2. We get rid of the environment variable, it gets from node store automatically, and if user wants to specify token they do it with a flag

Reply:
I would personally go with the second approach. If the user wants to use the token from the store but already has CELESTIA_NODE_AUTH_TOKEN in the env we will anyway get it from the env(even with the passed store flag). So, I would leave CELESTIA_NODE_AUTH_TOKEN as a fallback solution for getting the token.

Reply:
@vgonkivs @jcstein @distractedm1nd 

I think the cleanest approach is adding the `--p2p.network` command to rpc (in case it hasn't already) as well as the `--node.store` flag in the case there's a custom store, **AND** also add the `--auth.token` flag that allows user to pass token that will override the need for passing a network or node store path as the token will already have been passed. 

So from user perspective, it would look like this: 

**mocha** (default network):
```
celestia blob submit [params] # generates token from default mocha store
```

**arabica**: 
```
celestia blob submit [params] --p2p.network arabica # generates token from default arabica store
```

**custom node store**: 
```
celestia blob submit [params] --node.store <path to node store> # generates token from the given node store
```
**with auth token**: 
```
celestia blob submit [params] --auth.token <token> # takes the given token, no need to look in store
```

_**There is no more need for the env var at all in this case.**_

Reply:
With the above approach ^ there is no need for env var at all, it will only add confusion to user + difficulty to parsing logic

Reply:
In case of multiple calls, the user will have to specify auth.token/path each time which is weird imo. And having env solves this problem

Reply:
@renaynay, besides the network flag, the user will also have to pass node type flag

Reply:
```
celestia blob submit [params] --p2p.network arabica --node.type light # generates token from default arabica store
```

Reply:
```
celestia share get-share "$(celestia header get-by-height 147105 | jq '.result.dah' -r)" 0 2 --p2p.network arabica --node.type light
```

Reply:
queries could be more difficult than just submit ^

Reply:
ah yes node.type would be necessary too in case of just p2p.network. 

hm.

env var likely would be the best UX for repeated requests.